include/thread_pool/slotted_bag.hpp

```diff
@@ -0,0 +1,169 @@
+#pragma once
+
+#include <thread_pool/mpmc_bounded_queue.hpp>
+
+#include <atomic>
+
+namespace tp
+{
+
+/**
+* @brief The Slotted Bag is a lockless container which tracks membership by id. 
+* The bag is divided into slots, each with a unique id, and each slot can be full or empty.
+* The bag supports filling and emptying slots, as well as the unordered emptying of any full slot. 
+* The bag supports multiple consumers, and a single producer per slot.
+*/
+template <template<typename> class Queue>
+class SlottedBag
```

Comment:
This is a really niche structure I made up for keeping track of idle threads in the context of the lockfree interaction between workers and calls to `post()`. See the @brief above.